### PR TITLE
feat(amf): Handling security mode reject 

### LINKED ISSUE #7323: Security capabilities mismatch support is not added in AMF

### PR DESCRIPTION
**Pull request by Chandhu-Wavelabs**
Signed-off-by: chandhu-wavelabs <chandhu.naga@wavelabs.ai>

## Summary

Support added for handling security mode reject message

## Test Plan

unit test case is added for security mode reject decoding.
Tested the handling of security mode reject on UERANSIM setup

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp [hunk 1]
```diff
@@ -205,7 +205,13 @@ int amf_app_handle_deregistration_req(amf_ue_ngap_id_t ue_id) {
...
   }
   // UE context release notification to NGAP
-  amf_app_ue_context_release(ue_context, ue_context->ue_context_rel_cause);
+  if (ue_context->ue_context_rel_cause.ngapCause_u.nas ==
+      ngap_CauseNas_normal_release) {
+    ue_context->ue_context_rel_cause.ngapCause_u.nas = ngap_CauseNas_deregister;
+    amf_app_ue_context_release(ue_context, ue_context->ue_context_rel_cause);
+  } else {
+    amf_app_ue_context_release(ue_context, ue_context->ue_context_rel_cause);
+  }
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]:
```suggestion
  if (ue_context->ue_context_rel_cause.ngapCause_u.nas ==
      ngap_CauseNas_normal_release) {
    ue_context->ue_context_rel_cause.ngapCause_u.nas = ngap_CauseNas_deregister;
  } 
  amf_app_ue_context_release(ue_context, ue_context->ue_context_rel_cause);
  
```

Comment (ardzoht) [reply lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: This can be refactored

**Review by ardzoht — CHANGES_REQUESTED** [review]
Minor refactor to apply

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp (hunk 1 above)
Comment (Chandhu-Wavelabs) [reply lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: Done

**Review by ardzoht — APPROVED** [review]

**Comment by ardzoht** [comment]
Looks like Jenkins job is failing

File: lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp (hunk 1 above)
Comment (ulaskozat) [reply lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: I do not see any changes here with respect to Alex's comment. How is it `Done`?

Comment (Chandhu-Wavelabs) [reply lte/gateway/c/core/oai/tasks/amf/deregistration_request.cpp]: Fixed, I think these refactor changes were reverted while rebasing latest master.

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**